staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -131,21 +131,27 @@ func (rt *Transport) WrappedRoundTripper() http.RoundTripper {
 
 // rewriteURL rewrites a single URL to go through the proxy, if the URL refers
 // to the same host as sourceURL, which is the page on which the target URL
-// occurred. If any error occurs (e.g. parsing), it returns targetURL.
-func (t *Transport) rewriteURL(targetURL string, sourceURL *url.URL) string {
+// occurred, or if the URL matches the sourceHost. If any error occurs (e.g.
+// parsing), it returns targetURL.
+func (t *Transport) rewriteURL(targetURL string, sourceURL *url.URL, sourceHost string) string {
 	url, err := url.Parse(targetURL)
 	if err != nil {
 		return targetURL
 	}
 
-	isDifferentHost := url.Host != "" && url.Host != sourceURL.Host
+	isDifferentHost := url.Host != "" && url.Host != sourceURL.Host && url.Host != sourceHost
```

Comment:
This is difficult to reason about if sourceURL.Host and sourceHost are both non-empty… would it make sense to collapse to a single sourceHost param to this function?

Reply:
@liggitt Sorry I didn't get the idea here. I think there are cases where sourceURL.Host and sourceRequestHost are both non-empty, and we need to examine both of them to tell if the targetURL is in a different host.